assignments/array-methods.js

```diff
@@ -55,29 +55,67 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
-let fullName = [];
-console.log(fullName);
+let fullName= []
+runners.forEach((runner) => {
+  fullName.push(`${runner.first_name}  ${runner.last_name}`);
+})
+
+console.log(fullName)
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
+runners.map((item) => {
+  allCaps.push(item['first_name'].toUpperCase());
+})
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+runners.filter(item => {
+  if(item['shirt_size'] === "L"){
+    largeShirts.push(item['first_name'] + ' ' + item['last_name']);
+  }
+})
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
-console.log(ticketPriceTotal);
+//let ticketPriceTotal = [];
+let ticketPriceTotal = runners.reduce((accum, val) => {
+    return accum + val.donation;
+  }, 0);
+  console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
-// Problem 1
+// Problem 1  check to see if the runner's first name starts with the letter "M"
+let firstNames = [];
+runners.filter((runner) => {
+  if(runner['first_name'].charAt(0) === "M"){
+    firstNames.push(runner['first_name'])
+  }
+})
+console.log(firstNames)
+
+// Problem 2 check to see if the email ends with .com 
+let email = []
+runners.filter((runner) => {
+  if(runner['email'].slice(-4) === '.com'){
+     email.push(`${runner['first_name']} ${runner['last_name']} : ${runner['email']}`);
+  }
+})
+console.log(email)
+
+// Problem 3 check for companies that donated more than 100
+let company = [];
 
-// Problem 2
+runners.filter(runner => {
+  if(runner['donation'] >= 100){
+    company.push(`${runner['company_name']}: ${runner['donation']}`)
+  }
+})
 
-// Problem 3
\ No newline at end of file
+console.log(company)
```

Comment:
Good job with all these problems, try to refactor them all so they don't use .push()